metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -234,14 +234,32 @@ def _build_derived_metric_output_node(
             ),
             metric_specs=[metric_spec],
         )
+        # For nested ratio / derived metrics with time offset, apply offset after metric computation.
+        join_to_time_spine_node: Optional[JoinToTimeSpineNode] = None
+        if metric_spec.offset_window or metric_spec.offset_to_grain:
```

Comment:
What happens if there's a filter on metric_time?

Today we generally do:

1. join
2. filter
3. aggregate

All of this is in ComputeMetricsNode, which comes post-aggregation, so I think we'll end up joining the time spine against the filtered input set here, effectively truncating the input data on the second offset.

If I'm right about this, we'll need to propagate the offset information so we can expand that window. For now it might be adequate to apply the filter expression after the final calculation if there's a nested derived offset, and then we can optimize the predicate handling later.

Reply:
Moving this discussion over to [a new PR](https://github.com/dbt-labs/metricflow/pull/886), based off of one of Paul's branches